### PR TITLE
[33864] Add subproject boards and extract common board action class

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/33864

### CODE REVIEW

File: frontend/src/app/modules/boards/boards-root/boards-root.component.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import {BoardStatusActionService} from "core-app/modules/boards/board/board-acti
 import {BoardVersionActionService} from "core-app/modules/boards/board/board-actions/version/version-action.service";
 import {QueryUpdatedService} from "core-app/modules/boards/board/query-updated/query-updated.service";
 import {BoardAssigneeActionService} from "core-app/modules/boards/board/board-actions/assignee/assignee-action.service";
+import {BoardSubprojectActionService} from "core-app/modules/boards/board/board-actions/subproject/subproject-action.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/boards-root/boards-root.component.ts:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/ae626eab2c7b67611193ce7f8a8aa639c2e9156d/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L9"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/boards/board/board-partitioned-page/board-list-container.component.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ import {AddListModalComponent} from "core-app/modules/boards/board/add-list-moda
 import {I18nService} from "core-app/modules/common/i18n/i18n.service";
 import {BoardListCrossSelectionService} from "core-app/modules/boards/board/board-list/board-list-cross-selection.service";
 import {filter} from "rxjs/operators";
+import {BoardActionsRegistryService} from "core-app/modules/boards/board/board-actions/board-actions-registry.service";
 import {APIV3Service} from "core-app/modules/apiv3/api-v3.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/board/board-partitioned-page/board-list-container.component.ts:25]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/ae626eab2c7b67611193ce7f8a8aa639c2e9156d/frontend/src/app/modules/boards/board/board-partitioned-page/board-list-container.component.ts#L25"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/ae626eab2c7b67611193ce7f8a8aa639c2e9156d/frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L42">frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L42</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by ulferts — APPROVED** [review]
The code looks good and the functionality seems good enough although I have spotted some edge cases that are not intrinsic to this change but might diminish a user's experience when working with a subproject board nevertheless:

* A required custom field in one project will lead to the user being unable to move work packages into that project as the user is not offered any way to address the error within the board.
* A type not being available in one project will lead to a work package of that type being moved into that project switching the type "magically"
* Custom field getting lost without the user noticing
* For a project without subprojects, a newly created board is empty without the user being informed on what to do with the board and without the user being able to change the situation on the boards view.

Nevertheless, the code works and the shortcomings can be worked on indenpendently.

**Pull request merged**